conmon-rs/server/src/attach.rs

```diff
@@ -192,74 +203,96 @@ impl Attach {
         }
     }
 
-    async fn read_loop(mut read_half: OwnedReadHalf, tx: Sender<Vec<u8>>) -> Result<()> {
+    async fn read_loop(
+        mut read_half: OwnedReadHalf,
+        tx: Sender<Vec<u8>>,
+        token: CancellationToken,
+    ) -> Result<()> {
         loop {
             let mut buf = vec![0; Self::PACKET_BUF_SIZE];
-            match read_half.read(&mut buf).await {
-                Ok(n) if n > 0 => {
-                    if let Some(first_zero_idx) = buf.iter().position(|&x| x == 0) {
-                        buf.resize(first_zero_idx, 0);
+            select! {
```

Comment:
Beware of `select!` https://blog.yoshuawuyts.com/futures-concurrency-3/

I think in this case it may be OK because we'll never alternate between the two cases - if cancellation token fires, we're done.

But, I started adding "safety" comments to all my uses of `select!` after reading that blog.